sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,436 @@
+        >>> b.boundary_conditions['moment']
+        [(0, 4), (4, 0)]
+
+        """
+        if moment:
+            self._boundary_conditions['moment'].extend([m_bcs for m_bcs in moment])
+        if slope:
+            self._boundary_conditions['slope'].extend([s_bcs for s_bcs in slope])
+        if deflection:
+            self._boundary_conditions['deflection'].extend([d_bcs for d_bcs in deflection])
+
+    def apply_moment_boundary_conditions(self, *m_bcs):
+        """
+        Takes only the moment boundary conditions as input.
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_boundary_conditions(moment=[(0, 4), (4, 0)], deflection=[(0, 2)], slope=[(0, 1)])
+        >>> b.boundary_conditions
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+        >>> b.apply_moment_boundary_conditions((4, 3), (5, 0))
+        >>> b.boundary_conditions['moment']
+        [(0, 4), (4, 0), (4, 3), (5, 0)]
+
+        """
+        for bcs in m_bcs:
+            self._boundary_conditions['moment'].append(bcs)
+
+    def apply_slope_boundary_conditions(self, *s_bcs):
+        """
+        Takes only the slope boundary conditions as input.
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_boundary_conditions(moment=[(0, 4), (4, 0)], deflection=[(0, 2)], slope=[(0, 1)])
+        >>> b.boundary_conditions
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+        >>> b.apply_slope_boundary_conditions((4, 3), (5, 0))
+        >>> b.boundary_conditions['slope']
+        [(0, 1), (4, 3), (5, 0)]
+
+        """
+        for bcs in s_bcs:
+            self._boundary_conditions['slope'].append(bcs)
+
+    def apply_deflection_boundary_conditions(self, *d_bcs):
+        """
+        Takes only the slope boundary conditions as input.
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_boundary_conditions(moment=[(0, 4), (4, 0)], deflection=[(0, 2)], slope=[(0, 1)])
+        >>> b.boundary_conditions
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+        >>> b.apply_deflection_boundary_conditions((4, 3), (5, 0))
+        >>> b.boundary_conditions['deflection']
+        [(0, 2), (4, 3), (5, 0)]
+
+        """
+        for bcs in d_bcs:
+            self._boundary_conditions['deflection'].append(bcs)
+
+    def apply_load(self, value, start, order):
```

Comment:
We should have an optional `end` kwarg here that will let people define and end point for the distributed loads.